lib/URI/_query.pm

```diff
@@ -179,4 +180,15 @@ sub query_form_hash {
     return \%hash;
 }
 
+sub _is_array {
+    return(
+        defined($_[0]) &&
+        ( Scalar::Util::reftype($_[0]) || '' ) eq "ARRAY" && 
+        !(
+            Scalar::Util::blessed( $_[0] ) && 
+            overload::Method( $_[0], '""' )
```

Comment:
As far as I can tell, the tests never get to a scenario where `overload::Method( $_[0], '""' )` returns `true`. It would be helpful to allow for that case as well.